File: engines/java/src/main/java/com/dbloadbench/Inserter.java
public interface Inserter extends AutoCloseable {
    int defaultInsert(String csvFile, String tableName) throws Exception;
    int bulkInsert(String csvFile, String tableName, int batchSize) throws Exception;
    int fileInsert(String csvFile, String tableName) throws Exception;

    /**
     * Возвращает фактическое число строк в таблице через {@code SELECT COUNT(*)}.
     * Используется в Main для независимой проверки результата вставки — число
     * берётся из самой БД, а не из CSV, не из счётчика драйвера и не из
     * счётчика итераций.
     */
    int countRows(String tableName) throws Exception;

    void close();
}

File: engines/rust/src/inserter.rs
use anyhow::Result;

pub struct ConnParams {
    pub host:     String,
    pub port:     u16,
    pub user:     String,
    pub password: String,
    pub database: String,
}

pub trait Inserter {
    fn default_insert(&mut self, csv_file: &str, table: &str) -> Result<usize>;
    fn bulk_insert(&mut self, csv_file: &str, table: &str, batch_size: usize) -> Result<usize>;
    fn file_insert(&mut self, csv_file: &str, table: &str) -> Result<usize>;

    /// Возвращает фактическое число строк в таблице через `SELECT COUNT(*)`.
    /// Используется в `main` для независимой проверки результата вставки —
    /// число берётся именно из БД, а не из CSV и не из счётчиков драйвера.
    fn count_rows(&mut self, table: &str) -> Result<usize>;
}